src/Plugin/GraphQL/Schema/SdlSchemaPluginBase.php

```diff
@@ -222,10 +222,15 @@ protected function getSchemaDefinition() {
     $id = $this->getPluginId();
     $definition = $this->getPluginDefinition();
     $module = $this->moduleHandler->getModule($definition['provider']);
-    $file = "{$module->getPath()}/graphql/{$id}.graphqls";
+    $path = 'graphql/' . $id . '.graphqls';
+    $file = $module->getPath() . '/' . $path;
 
     if (!file_exists($file)) {
-      throw new InvalidPluginDefinitionException(sprintf("Missing schema definition file at %s.", $file));
+      throw new InvalidPluginDefinitionException(
+        $id,
+	sprintf(
+          'The module "%s" needs to have a schema definition "%s" in its folder for "%s" to be valid.',
+          \Drupal::moduleHandler()->getName($module->getName()), $path, $definition['class']));
```

Comment:
any reason we need the module handler here? We should avoid global \Drupal calls in classes and module->getName() should be enough here?

Reply:
This is for getting the human readable module name. If the machine name is enough, then we can simply use `$module->getName()` instead of `\Drupal::moduleHandler()->getName($module->getName())`.

Reply:
yep exactly, let's use the machine name. This is a developer facing exception, so should be fine.